src/mame/drivers/pc88va.cpp

```diff
@@ -1573,14 +1577,14 @@ void pc88va_state::dma_memw_cb(offs_t offset, uint8_t data)
 
 void pc88va_state::pc88va(machine_config &config)
 {
-	V50(config, m_maincpu, 8000000); // μPD9002, aka V30 + μPD70008AC (for PC8801 compatibility mode)
+	V50(config, m_maincpu, MASTER_CLOCK); // μPD9002, aka V30 + μPD70008AC (for PC8801 compatibility mode)
 	m_maincpu->set_addrmap(AS_PROGRAM, &pc88va_state::pc88va_map);
 	m_maincpu->set_addrmap(AS_IO, &pc88va_state::pc88va_io_map);
 	m_maincpu->set_vblank_int("screen", FUNC(pc88va_state::pc88va_vrtc_irq));
 	m_maincpu->set_irq_acknowledge_callback("pic8259_master", FUNC(pic8259_device::inta_cb));
 
 #if TEST_SUBFDC
-	z80_device &fdccpu(Z80(config, "fdccpu", 8000000));        /* 8 MHz */
+	z80_device &fdccpu(Z80(config, "fdccpu", MASTER_CLOCK));        /* 8 MHz */
```

Comment:
This is 4 MHz, it’s a PC-80S31K or whatever internal/external variety. 
PS: Arguably the hookup can be converted to use the device itself now but iirc “intelligent mode” is for PC-8801 compatibility SWs anyway.

Reply:
So you mean it *should* be 4 MHz despite what the comment says?

Or did you think that `MASTER_CLOCK` was defined as 4 MHz? (It is defined to `XTAL(8'000'000)` in pc88va.cpp.)

Reply:
It’s an independent device, running on a separate xtal. PC-88VA is most likely internal, but it was originally an external box.
![image](https://user-images.githubusercontent.com/9251271/147692433-36d053e2-f0ed-49e1-9f17-61becd95c593.jpeg)

Reply:
https://github.com/mamedev/mame/blob/6ffeb38cacc43aca30b7d1468b6f2407635d6d2c/src/mame/machine/pc80s31k.cpp#L40

Reply:
I reverted the changes related to the floppy device for now.

Going by the discussion from #8560 we shouldn't change it from 8 MHz to 4 MHz unless we have proof that it really runs at 4 MHz or 3.99 MHz.